api/v1beta1/glance_webhook.go

```diff
@@ -23,19 +23,56 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/webhook"
 )
 
+// GlanceDefaults -
+type GlanceDefaults struct {
+	ContainerImageURL string
+}
+
+var glanceDefaults GlanceDefaults
+
 // log is for logging in this package.
 var glancelog = logf.Log.WithName("glance-resource")
 
 // SetupWebhookWithManager sets up the webhook with the Manager
-func (r *Glance) SetupWebhookWithManager(mgr ctrl.Manager) error {
+func (r *Glance) SetupWebhookWithManager(mgr ctrl.Manager, defaults GlanceDefaults) error {
+	glanceDefaults = defaults
+
 	return ctrl.NewWebhookManagedBy(mgr).
 		For(r).
 		Complete()
 }
 
-// TODO(user): EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+//+kubebuilder:webhook:path=/mutate-glance-openstack-org-v1beta1-glance,mutating=true,failurePolicy=fail,sideEffects=None,groups=glance.openstack.org,resources=glances,verbs=create;update,versions=v1beta1,name=mglance.kb.io,admissionReviewVersions=v1
+
+var _ webhook.Defaulter = &Glance{}
+
+// Default implements webhook.Defaulter so a webhook will be registered for the type
+func (r *Glance) Default() {
+	glancelog.Info("default", "name", r.Name)
+
+	r.Spec.Default(glanceDefaults)
+}
+
+// Default - set defaults for this Glance spec
+func (spec *GlanceSpec) Default(defaults GlanceDefaults) {
```

Comment:
I'm ambivalent on whether having the `defaults GlanceDefaults` arg here is worthwhile or not, but I just thought I'd include it initially to see what people thought